### ISSUE TITLE
Cisco Mibs Brake when using mib names

### ISSUE THREAD

**Issue by seanfariacustomink** [swisscom/ruby-netsnmp#72]
If I use the Cisco mibs by name (i.e. ciscoMemoryPoolFree), I would get the following error:
```
"errorMessage": "invalid OBJECT ID 48.1.1.1.7.7: first num too large",
"errorType": "Function<OpenSSL::ASN1::ASN1Error>"
```
If I use the OID it works and provides the correct feedback

**Comment by HoneyryderChuck** [comment]
Thx for the report.

Is the oid from the error message correct? What's the difference in the logs when you turn on the debug output?

**Comment by seanfariacustomink** [comment]
Yes the mib is correct, but it seems the first part of it gets cut off see below:

OID = 1.3.6.1.4.1.9.9.48.1.1.1.6 = CISCO-MEMORY-POOL-MIB::ciscoMemoryPoolFree

--

**Comment by HoneyryderChuck** [comment]
That looks like a mib parsing issue. Do you think you can write a patch to add support for that case? Or alternatively,  can you upload the cisco mibfile,  so I can do it when I have the time?

**Comment by seanfariacustomink** [comment]
Sorry still a newbie, I can give it a shot.

List of cisco mibs:
https://github.com/cisco/cisco-mibs/tree/main/v2
Actual mib in question:
https://github.com/cisco/cisco-mibs/blob/main/v2/CISCO-MEMORY-POOL-MIB.my

**Comment by HoneyryderChuck** [comment]
thx, I'll see what I can do.

**Comment by HoneyryderChuck** [comment]
It seems that you are missing a MIB file in the mibdir, particularly [this one](https://github.com/cisco/cisco-mibs/blob/main/v2/CISCO-SMI.my). I could reproduce it by doing:

```ruby
require "netsnmp"
require "netsnmp/mib/parser"

NETSNMP::MIB.load(File.join(__dir__, "CISCO-SMI.my"))
NETSNMP::MIB.load(File.join(__dir__, "CISCO-MEMORY-POOL-MIB.txt"))

puts NETSNMP::MIB.oid("ciscoMemoryPoolFree")
# => 1.3.6.1.4.1.9.9.48.1.1.1.6
```

Can you confirm that you're missing the file?

**Comment by seanfariacustomink** [comment]
Interesting!!! I did have it like this:
```
  NETSNMP::MIB.load('./CISCO-MEMORY-POOL-MIB.my')
  NETSNMP::MIB.load('./CISCO-PROCESS-MIB.my')
  NETSNMP::MIB.load('./CISCO-PROCESS-CAPABILITY.my')
  NETSNMP::MIB.load('./CISCO-REMOTE-ACCESS-MONITOR-MIB.my')
  NETSNMP::MIB.load('./CISCO-FIREWALL-MIB.my')
  NETSNMP::MIB.load('./CISCO-QOS-PIB-MIB.my')
  NETSNMP::MIB.load('./CISCO-SMI.my')
  NETSNMP::MIB.load('./powernet441.mib')
```
hmm let me try to load the entire directory some how, but the error does change when I use the following:
 ```
NETSNMP::MIB.load(File.join(__dir__, "CISCO-SMI.my"))
NETSNMP::MIB.load(File.join(__dir__, "CISCO-MEMORY-POOL-MIB.txt"))
```

Let me try this and get back to you today.

**Comment by seanfariacustomink** [comment]
After changing the all of the above to `NETSNMP::MIB.load(File.join(__dir__, "MIB"))` format it states the following error:

`"errorMessage": "Failed to match sequence (SPACE{0, } MODULES?) at line 16 char 1.",
"errorType": "Function<Parslet::ParseFailed>",`

**Comment by HoneyryderChuck** [comment]
so if I understood correctly, there was an error parsing the "CISCO-PROCESS-MIB.my" file?

**Comment by seanfariacustomink** [comment]
Yes that's correct

**Comment by HoneyryderChuck** [comment]
can you try with the master branch? I'm able to load those files already. Perhaps this has been fixed with my last week's patch.

**Comment by HoneyryderChuck** [comment]
@seanfariacustomink any updates?

**Comment by seanfariacustomink** [comment]
Hey, @HoneyryderChuck sorry for the delay got sick. Now i can't use any Cisco mibs name or oid.
With oid:
`  "errorMessage": "cannot load such file -- netsnmp",
  "errorType": "Init<LoadError>",`
  
  With Name:
`"errorMessage": "cannot load such file -- netsnmp",
"errorType": "Init<LoadError>",`

**Comment by HoneyryderChuck** [comment]
did you blindly copy-paste my statement above? because it contains a type (`netsnmp`, not `netsmp`).

**Comment by seanfariacustomink** [comment]
no I did not:
![image](https://user-images.githubusercontent.com/106612499/224748272-a84da446-de68-4363-ad4f-e74a33f5363b.png)
![image](https://user-images.githubusercontent.com/106612499/224748568-74464bc5-2773-45ba-9634-e4fad4d70d87.png)

**Comment by HoneyryderChuck** [comment]
apologies for the delay, seems I missed the notification.

Nothing strikes me out of the ordinary for that error to happen.
